blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md

```diff
@@ -176,14 +221,18 @@ If the machine running `exchangeRatesWorkflow()` crashes, Temporal can resume th
 
 The `exchangeRatesWorkflow` can run for an unlimited period of time: days, months, even years.
 However, Temporal caps a Workflow at 50,000 events. (See the [Time constraints](https://docs.temporal.io/workflows#time-constraints) section in [Temporal Workflows](https://docs.temporal.io/workflows).)
-In the `exchangeRatesWorkflow`, four events are fired at every iteration of the `while` loop, assuming no API errors.
+In the `exchangeRatesWorkflow`, eight events are fired at every iteration of the `while` loop, assuming no API errors.
 
 1. `EVENT_TYPE_TIMER_FIRED`: the `setTimeout()` resolved and it's time to refresh the exchange rates
-2. `EVENT_TYPE_ACTIVITY_TASK_STARTED`: the `getExchangeRates()` activity started executing
-3. `EVENT_TYPE_ACTIVITY_TASK_COMPLETED`: the `getExchangeRates()` activity completed successfully
-4. `EVENT_TYPE_TIMER_STARTED`: the Workflow used `setTimeout()` to pause until tomorrow
-
-With one API request per day, the `exchangeRatesWorkflow()` can run for almost 12,500 days (approximately 34 years) before running into the 50,000 event limit.
+2. `EVENT_TYPE_ACTIVITY_TASK_SCHEDULED`: the Temporal server scheduled the `getExchangeRates()` activity
+3. `EVENT_TYPE_ACTIVITY_TASK_STARTED`: the `getExchangeRates()` activity started executing
+4. `EVENT_TYPE_ACTIVITY_TASK_COMPLETED`: the `getExchangeRates()` activity completed successfully
+5. `EVENT_TYPE_WORKFLOW_TASK_SCHEDULED`: the Temporal server scheduled the Workflow logic that handles `getExchangeRates()`
+6. `EVENT_TYPE_WORKFLOW_TASK_STARTED`: the Workflow resumed
+7. `EVENT_TYPE_WORKFLOW_TASK_COMPLETED`: the Workflow paused
+8. `EVENT_TYPE_TIMER_STARTED`: the Workflow used `setTimeout()` to pause until tomorrow
+
+With one API request per day, the `exchangeRatesWorkflow()` can run for almost 6,250 days (approximately 17 years) before running into the 50,000 event limit.
```

Comment:
My recent understanding is that in reality we want to keep it under 500.

Reply:
Event history length under 500? Due to replay time?